### ISSUE TITLE
PoliteDesire constraints question

### ISSUE THREAD

**Issue by abelthefirst** [ignaciotcrespo/MrPolite#2]
```
import com.github.ignaciotcrespo.mrpolite.MrPolite;

public class Main {

    private interface ITest {
    }

    private static class ITestImpl implements ITest {

        @Override
        public String toString() {
            return "ololo";
        }
    }

    private static class Test {

        ITest str;

    }

    public static void main(final String... args) {
        Test test = MrPolite.one(Test.class).withClassEqualTo(ITest.class, new ITestImpl()).please();

        System.out.println(test.str);
    }

}
```

I was expecting "ololo" instead of null, am I getting the idea of withClassEqualTo() wrong?

**Comment by ignaciotcrespo** [comment]
hi Abel! Thanks for checking, will create a UT for your case and fix it. The current snapshot version supports interfaces, it creates a proxy and return random values for the methods, it will be released soon.

**Comment by ignaciotcrespo** [comment]
hey Abel! The bug seems to be fixed in v1.0.5, check commit https://github.com/ignaciotcrespo/MrPolite/commit/1f868fe8fd5547638953fe09fab26b450ea2805a 😉 

I released v1.0.5, usually it takes half hour to be available in maven.

A big hug for you man.
